ext/mbstring/mbstring.c

```diff
@@ -3068,55 +3065,77 @@ static const mbfl_encoding* mb_guess_encoding(unsigned char *in, size_t in_len,
 					if (w == MBFL_BAD_INPUT) {
 						if (strict) {
 							/* This candidate encoding is not valid, eliminate it from consideration */
-							elist_size--;
-							memmove(&data[i], &data[i+1], (elist_size - i) * sizeof(struct conversion_data));
+							length--;
+							if (length == 0) {
+								return 0;
+							}
+							memmove(&array[i], &array[i+1], (length - i) * sizeof(struct candidate));
 							goto try_next_encoding;
 						} else {
-							data[i].demerits += 1000;
+							array[i].demerits += 1000;
 						}
 					} else {
-						data[i].demerits += mb_estimate_encoding_demerits(w);
+						array[i].demerits += estimate_demerits(w);
 					}
 				}
-				if (data[i].in_len == 0) {
+				if (array[i].in_len == 0) {
 					finished++;
 				}
 			}
-			i++;
 		}
 	}
 
-	if (strict) {
+	return length;
+}
+
+MBSTRING_API const mbfl_encoding* mb_guess_encoding_for_strings(unsigned char **strings, size_t *str_lengths, size_t n, const mbfl_encoding **elist, unsigned int elist_size, bool strict)
+{
+	if (elist_size == 0) {
+		return NULL;
+	}
+	if (elist_size == 1) {
+		if (strict) {
+			while (n--) {
+				if (!php_mb_check_encoding((const char*)strings[n], str_lengths[n], *elist)) {
```

Comment:
Can't we prevent the const cast by requesting ``const unsigned char **strings`` ?